src/components/year_month.rs

```diff
@@ -61,6 +68,80 @@ impl<C: CalendarProtocol> YearMonth<C> {
     pub fn calendar(&self) -> &CalendarSlot<C> {
         &self.calendar
     }
+
+    /// Returns the Calendar value.
+    #[inline]
+    #[must_use]
+    pub fn in_leap_year(&self) -> bool {
+        utils::mathematical_in_leap_year(utils::epoch_time_for_year(self.iso.year)) == 1
+    }
+}
+
+// Contextual Methods
+impl<C: CalendarProtocol> YearMonth<C> {
+    pub fn contextual_get_days_in_year(
+        this: &C::YearMonth,
+        context: &mut C::Context,
+    ) -> TemporalResult<u16> {
+        this.get_calendar()
+            .days_in_year(&CalendarDateLike::YearMonth(this.clone()), context)
+    }
+
+    pub fn contextual_get_days_in_month(
+        this: &C::YearMonth,
+        context: &mut C::Context,
+    ) -> TemporalResult<u16> {
+        this.get_calendar()
+            .days_in_month(&CalendarDateLike::YearMonth(this.clone()), context)
+    }
+
+    pub fn contextual_month_code(
+        this: &C::YearMonth,
+        context: &mut C::Context,
+    ) -> TemporalResult<TinyAsciiStr<4>> {
+        this.get_calendar()
+            .month_code(&CalendarDateLike::YearMonth(this.clone()), context)
+    }
+
+    pub fn contextual_get_months_in_year(
+        this: &C::YearMonth,
+        context: &mut C::Context,
+    ) -> TemporalResult<u16> {
+        this.get_calendar()
+            .months_in_year(&CalendarDateLike::YearMonth(this.clone()), context)
+    }
+
+    pub fn contextual_add_or_subtract_duration(
+        operation: DurationOperation,
+        this: &C::YearMonth,
+        mut duration: Duration,
+        context: &mut C::Context,
+        overflow: ArithmeticOverflow,
+    ) -> TemporalResult<YearMonth<C>> {
+        match operation {
+            DurationOperation::Subtract => duration = duration.negated(),
+            DurationOperation::Add => {}
+        }
+
+        let mut fields = TemporalFields::default();
```

Comment:
thought: huh, seeing this makes me wonder if it would be worthwhile to implement a  `From<YearMonth> for TemporalFields` and `From<Date> for TemporalFields`

Reply:
its difficult to do this as this is an associated type `From<&<C as CalendarProtocol>::YearMonth>`, i couldn't find a way to make this work

Reply:
Follow up! So I think the best approach would be below. Found it when working the new `Date` methods. :smile: 

```rust

let fields = TemporalFields {
  bit_map: FieldMap::YEAR | FieldMap::MONTH,
  year: Some(this.iso_date().year),
  month: Some(this.iso_date().month),
  ..Default::default(),
}

```
Also, there may be an argument for implementing the from as a tuple, `impl From<(i32, u8)> for TemporalFields`. That might be able to be extrapolated out a bit further with a wrapper to ensure it's being converted properly (granted this also may be doing too much). 

```rust

struct YearMonthFields((i32, u8));

impl From<YearMonthFields> for TemporalFields {
  fn from(value: YearMonthFields) -> Self {
    TemporalFields {
      bit_map: FieldMap::YEAR | FieldMap::MONTH,
      year: Some(value.0.0),
      month: Some(value.0.1),
      ..Default::default(),
    }
  }
}

```

`MonthDay` could then be very similar with (u8, u8) in that instance. Let me know what you think!

Reply:
Yep I like this idea, looks great, thank you